### ISSUE TITLE
http status 500

### ISSUE THREAD

**Issue by brunoap** [sabeechen/hassio-google-drive-backup#225]
{"http_status": 500, "error_type": "generic_error", "message": "An unexpected error occurred: (22, 'Invalid argument')", "details": "\n  /site-packages/caio/asyncio_base.py:43 (step)\nSystemError: (22, 'Invalid argument')\nWhose handling caused:\n  addon/backup/ui/uiserver.py:608 (error_middleware)\n  addon/backup/ui/uiserver.py:317 (changefolder)\n  addon/backup/drive/folderfinder.py:81 (save)\n  /site-packages/aiofile/aio.py:211 (write)\n  [2 hidden frames]\n  /site-packages/caio/asyncio_base.py:48 (step)\nSystemError: (22, 'Invalid argument')\n"}

**Comment by brunoap** [comment]
> {"http_status": 500, "error_type": "generic_error", "message": "An unexpected error occurred: (22, 'Invalid argument')", "details": "\n /site-packages/caio/asyncio_base.py:43 (step)\nSystemError: (22, 'Invalid argument')\nWhose handling caused:\n addon/backup/ui/uiserver.py:608 (error_middleware)\n addon/backup/ui/uiserver.py:317 (changefolder)\n addon/backup/drive/folderfinder.py:81 (save)\n /site-packages/aiofile/aio.py:211 (write)\n [2 hidden frames]\n /site-packages/caio/asyncio_base.py:48 (step)\nSystemError: (22, 'Invalid argument')\n"}

Error happens after selecting google drive folder. 
Actualy, uninstalled and reinstalled because it was stucked on syncing. 
After reinstall this error started and i cannot select a folder on gdrive.

**Comment by brunoap** [comment]
###### Description:
```
If you have anything else that could help explain what happened, click "Markdown" above and write it here.
```

 Addon version: `0.101.0`
 Home Assistant Version: `0.114.1`
 Supervisor Version: `234`
 Architecture: `armv7`
 Date: `2020-08-15T17:43:51.297689+00:00`
 Timezone: `America/Sao_Paulo`
 Failure Time: `2020-08-15T17:43:44.239548+00:00`
 Last Good Sync: `2020-08-15T17:43:44.214061+00:00`
 ###### Exception:
 ```
 
  /site-packages/caio/asyncio_base.py:43 (step)
SystemError: (22, 'Invalid argument')
Whose handling caused:
  addon/backup/model/coordinator.py:148 (_sync)
  addon/backup/model/model.py:133 (sync)
  addon/backup/model/model.py:216 (_syncSnapshots)
  addon/backup/drive/drivesource.py:84 (get)
  addon/backup/drive/drivesource.py:187 (getFolderId)
  addon/backup/drive/folderfinder.py:52 (get)
  addon/backup/drive/folderfinder.py:81 (save)
  /site-packages/aiofile/aio.py:211 (write)
  [2 hidden frames]
  /site-packages/caio/asyncio_base.py:48 (step)
SystemError: (22, 'Invalid argument')

 ```
 Snapshots:
 ```
 {
    "HomeAssistant": {
        "snapshots": 1,
        "retained": 0,
        "deletable": 1,
        "name": "HomeAssistant",
        "latest": "2020-08-15T17:02:23Z",
        "size": "602.6 MB"
    },
    "GoogleDrive": {
        "snapshots": 0,
        "retained": 0,
        "deletable": 0,
        "name": "GoogleDrive",
        "latest": null,
        "size": "0.0 B"
    }
}
 ```
 ###### Config:
 ```
 {
    "Setting.DAYS_BETWEEN_SNAPSHOTS": 1.0
}
 ```
 ###### Addon Logs:
 ```
 08-15 14:43:15 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/addons/self/info
08-15 14:43:15 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/info
08-15 14:43:15 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/homeassistant/info
08-15 14:43:15 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/supervisor/info
08-15 14:43:15 INFO [backup.ui.uiserver] Starting server on port 8099
08-15 14:43:15 INFO [backup.ui.uiserver] Server started
08-15 14:43:15 DEBUG [backup.model.syncer] Sync requested by Coordinator
08-15 14:43:15 INFO [backup.model.coordinator] Syncing Snapshots
08-15 14:43:15 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
08-15 14:43:15 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots/27a2d344/info
08-15 14:43:33 INFO [backup.drive.drivesource] Saving new Google Drive credentials
08-15 14:43:33 DEBUG [backup.model.syncer] Sync requested by GoogleDrive
08-15 14:43:33 INFO [backup.model.coordinator] Syncing Snapshots
08-15 14:43:33 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
08-15 14:43:33 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=mimeType%3D%27application%2Fvnd.google-apps.folder%27&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
08-15 14:43:34 INFO [backup.drive.folderfinder] Found Hass.io Snapshots
08-15 14:43:34 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%271D8wYcx573Zfv8C7Bs7fhz5H9eFw2FKIC%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
08-15 14:43:34 ERROR [backup.model.coordinator] A backup folder already exists.  Please visit the add-on Web UI to select where to backup.
08-15 14:43:34 INFO [backup.model.coordinator] I'll try again in an hour
08-15 14:43:44 INFO [backup.model.coordinator] Syncing Snapshots
08-15 14:43:44 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
08-15 14:43:44 INFO [backup.drive.folderfinder] Saving snapshot folder: 1D8wYcx573Zfv8C7Bs7fhz5H9eFw2FKIC
08-15 14:43:44 ERROR [backup.model.coordinator] 
  /site-packages/caio/asyncio_base.py:43 (step)
SystemError: (22, 'Invalid argument')
Whose handling caused:
  addon/backup/model/coordinator.py:148 (_sync)
  addon/backup/model/model.py:133 (sync)
  addon/backup/model/model.py:216 (_syncSnapshots)
  addon/backup/drive/drivesource.py:84 (get)
  addon/backup/drive/drivesource.py:187 (getFolderId)
  addon/backup/drive/folderfinder.py:52 (get)
  addon/backup/drive/folderfinder.py:81 (save)
  /site-packages/aiofile/aio.py:211 (write)
  [2 hidden frames]
  /site-packages/caio/asyncio_base.py:48 (step)
SystemError: (22, 'Invalid argument')

08-15 14:43:44 INFO [backup.model.coordinator] I'll try again in an hour
 ```
 ###### Supervisor Logs:
 ```
 [32m20-08-15 17:28:54 INFO (SyncWorker_5) [supervisor.docker.interface] Remove image sabeechen/hassio-google-drive-backup-armv7 with latest and 3.12[0m
[32m20-08-15 17:28:54 INFO (MainThread) [supervisor.addons.addon] Remove add-on data folder /data/addons/data/cebe7a76_hassio_google_drive_backup[0m
[32m20-08-15 17:28:55 INFO (MainThread) [supervisor.addons] Add-on 'cebe7a76_hassio_google_drive_backup' successfully removed[0m
[31m20-08-15 17:29:23 ERROR (MainThread) [supervisor.snapshots] A snapshot/restore process is already running[0m
[32m20-08-15 17:31:33 INFO (MainThread) [supervisor.snapshots] Found 0 snapshot files[0m
[32m20-08-15 17:31:35 INFO (MainThread) [supervisor.snapshots] Found 0 snapshot files[0m
[32m20-08-15 17:31:49 INFO (MainThread) [supervisor.snapshots] Found 0 snapshot files[0m
[32m20-08-15 17:32:03 INFO (MainThread) [supervisor.snapshots] Found 0 snapshot files[0m
[32m20-08-15 17:33:48 INFO (SyncWorker_7) [supervisor.snapshots.snapshot] Snapshot folder homeassistant done[0m
[32m20-08-15 17:33:48 INFO (SyncWorker_4) [supervisor.snapshots.snapshot] Snapshot folder share[0m
[32m20-08-15 17:33:48 INFO (SyncWorker_4) [supervisor.snapshots.snapshot] Snapshot folder share done[0m
[32m20-08-15 17:33:48 INFO (SyncWorker_6) [supervisor.snapshots.snapshot] Snapshot folder addons/local[0m
[32m20-08-15 17:33:48 INFO (SyncWorker_6) [supervisor.snapshots.snapshot] Snapshot folder addons/local done[0m
[32m20-08-15 17:33:48 INFO (SyncWorker_3) [supervisor.snapshots.snapshot] Snapshot folder ssl[0m
[32m20-08-15 17:33:48 INFO (SyncWorker_3) [supervisor.snapshots.snapshot] Snapshot folder ssl done[0m
[32m20-08-15 17:35:08 INFO (MainThread) [supervisor.snapshots] Full-Snapshot 27a2d344 done[0m
[32m20-08-15 17:36:10 INFO (MainThread) [supervisor.api.snapshots] Download snapshot 27a2d344[0m
[32m20-08-15 17:42:23 INFO (MainThread) [supervisor.addons] Create Home Assistant add-on data folder /data/addons/data/cebe7a76_hassio_google_drive_backup[0m
[32m20-08-15 17:42:23 INFO (SyncWorker_3) [supervisor.docker.interface] Pull image sabeechen/hassio-google-drive-backup-armv7 tag 0.101.0.[0m
[32m20-08-15 17:42:26 INFO (MainThread) [supervisor.addons] Add-on 'cebe7a76_hassio_google_drive_backup' successfully installed[0m
[32m20-08-15 17:43:12 INFO (SyncWorker_4) [supervisor.docker.addon] Start Docker add-on sabeechen/hassio-google-drive-backup-armv7 with version 3.12[0m
[31m20-08-15 17:43:13 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.33.4:8099 ssl:default [Connection refused][0m
[32m20-08-15 17:43:15 INFO (MainThread) [supervisor.api.security] /homeassistant/info access from cebe7a76_hassio_google_drive_backup[0m
[32m20-08-15 17:43:15 INFO (MainThread) [supervisor.api.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m20-08-15 17:43:15 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m20-08-15 17:43:15 INFO (MainThread) [supervisor.api.security] /snapshots/27a2d344/info access from cebe7a76_hassio_google_drive_backup[0m
[32m20-08-15 17:43:33 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m20-08-15 17:43:44 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m20-08-15 17:43:51 INFO (MainThread) [supervisor.api.security] /supervisor/logs access from cebe7a76_hassio_google_drive_backup[0m

 ```

 ```

**Comment by sabeechen** [comment]
I've just released a new version, v0.101.1, that should resolve this.  I think one of the libraries I was using has compatibility problems on ARM platforms (like the pi), so I used something else.  If it doesn't work for you after updating, please let me know.
